From 4067d9fd903bf258a63f481c607d7a65fe0d9539 Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 22 Nov 2024 09:32:32 +0100
Subject: [PATCH 1367/1695] drivers: stm32_bsec3: rename bsec shadow to mirror

OTP shadowing has a different meaning than mirroring. Shadowing is the
copy of an OTP from the fuse bank to BSEC register. Mirroring is the
copy of shadowed OTP to a memory that can be accessed by other software
components. So rename in BSEC3 driver the wrongly used shadow with
mirror.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 89715ed4775a1b0b809a3c524d104edb8f13056e drivers: stm32_bsec3: rename bsec shadow to mirror]
Change-Id: I18527d66b0899d9d70932cca4aa114dda3aa8689
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/420239
Reviewed-by: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
Domain-Review: Thomas BOURGOIN <thomas.bourgoin@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/plat-stm32mp2/main.c |   2 +-
 core/drivers/stm32_bsec3.c         | 163 +++++++++++++++--------------
 core/include/drivers/stm32_bsec.h  |   4 +-
 3 files changed, 85 insertions(+), 84 deletions(-)

diff --git a/core/arch/arm/plat-stm32mp2/main.c b/core/arch/arm/plat-stm32mp2/main.c
index 84d9d08da..580628e83 100644
--- a/core/arch/arm/plat-stm32mp2/main.c
+++ b/core/arch/arm/plat-stm32mp2/main.c
@@ -178,7 +178,7 @@ void plat_abort_handler(struct thread_abort_regs *regs __unused)
 void plat_bsec_get_static_cfg(struct stm32_bsec_static_cfg *cfg)
 {
 	cfg->base = BSEC3_BASE;
-	cfg->shadow = SRAM1_BASE;
+	cfg->mirror = SRAM1_BASE;
 	cfg->upper_start = STM32MP2_UPPER_OTP_START;
 	cfg->max_id = STM32MP2_OTP_MAX_ID;
 }
diff --git a/core/drivers/stm32_bsec3.c b/core/drivers/stm32_bsec3.c
index eb2d936e4..8ceff9a2e 100644
--- a/core/drivers/stm32_bsec3.c
+++ b/core/drivers/stm32_bsec3.c
@@ -149,8 +149,8 @@ static_assert(!(IS_ENABLED(CFG_STM32_CM33TDCID) &&
 
 #define OTP_ACCESS_SIZE			12U
 
-#define HIDEUP_ERROR	(PTA_BSEC_LOCK_SHADOW_R | PTA_BSEC_LOCK_SHADOW_W | PTA_BSEC_LOCK_SHADOW_P |\
-			 PTA_BSEC_LOCK_ERROR)
+#define HIDEUP_ERROR	(PTA_BSEC_LOCK_SHADOW_R | PTA_BSEC_LOCK_SHADOW_W | \
+			 PTA_BSEC_LOCK_SHADOW_P | PTA_BSEC_LOCK_ERROR)
 
 /* OTP18 = BOOTROM_CONFIG_0-3: Security life-cycle word 2 */
 #define OTP_SECURE_BOOT			18U
@@ -164,26 +164,26 @@ struct nvmem_cell {
 	size_t bit_len;
 };
 
+/* Magic use to indicated valid mirror = 'B' 'S' 'E' 'C' */
+#define BSEC_MAGIC			0x42534543
+
+struct bsec_mirror {
+	uint32_t magic;
+	uint32_t state;
+	uint32_t value[OTP_MAX_SIZE];
+	uint32_t status[OTP_MAX_SIZE];
+};
+
 struct bsec_dev {
 	struct io_pa_va base;
 	unsigned int upper_base;
 	unsigned int max_id;
 	unsigned int lock;
-	struct bsec_shadow *shadow;
+	struct bsec_mirror *mirror;
 	struct nvmem_cell *cells;
 	size_t cell_count;
 };
 
-/* Magic use to indicated valid SHADOW = 'B' 'S' 'E' 'C' */
-#define BSEC_MAGIC			0x42534543
-
-struct bsec_shadow {
-	uint32_t magic;
-	uint32_t state;
-	uint32_t value[OTP_MAX_SIZE];
-	uint32_t status[OTP_MAX_SIZE];
-};
-
 /* Only 1 instance of BSEC is expected per platform */
 static struct bsec_dev bsec_dev = {
 	.lock = SPINLOCK_UNLOCK
@@ -257,14 +257,14 @@ static TEE_Result shadow_otp(unsigned int otp)
 	uint32_t status = 0U;
 
 	/* if shadow is not allowed */
-	if (bsec_dev.shadow->status[otp] & PTA_BSEC_LOCK_SHADOW_R) {
-		bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_ERROR;
-		bsec_dev.shadow->value[otp] = 0x0U;
+	if (bsec_dev.mirror->status[otp] & PTA_BSEC_LOCK_SHADOW_R) {
+		bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_ERROR;
+		bsec_dev.mirror->value[otp] = 0x0U;
 
 		return TEE_ERROR_GENERIC;
 	}
 
-	bsec_dev.shadow->status[otp] &= ~PTA_BSEC_LOCK_ERROR;
+	bsec_dev.mirror->status[otp] &= ~PTA_BSEC_LOCK_ERROR;
 	for (i = 0U; i < MAX_NB_TRIES; i++) {
 		io_write32(bsec_base() + BSEC_OTPCR, otp);
 
@@ -279,11 +279,11 @@ static TEE_Result shadow_otp(unsigned int otp)
 		break;
 	}
 	if (status & BSEC_OTPSR_PPLF)
-		bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_PERM;
+		bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_PERM;
 	if (i == MAX_NB_TRIES || status & (BSEC_OTPSR_PPLMF | BSEC_OTPSR_AMEF |
 					   BSEC_OTPSR_DISTURBF |
 					   BSEC_OTPSR_DEDF)) {
-		bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_ERROR;
+		bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_ERROR;
 		return TEE_ERROR_GENERIC;
 	}
 
@@ -342,9 +342,9 @@ TEE_Result stm32_bsec_read_otp(uint32_t *val, uint32_t otp)
 	if (otp > bsec_dev.max_id)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	/* for non-secure OTP: only return the shadow value */
-	if (!(bsec_dev.shadow->status[otp] & PTA_BSEC_STATUS_SECURE)) {
-		*val = bsec_dev.shadow->value[otp];
+	/* for non-secure OTP: only return the mirror value */
+	if (!(bsec_dev.mirror->status[otp] & PTA_BSEC_STATUS_SECURE)) {
+		*val = bsec_dev.mirror->value[otp];
 
 		return TEE_SUCCESS;
 	}
@@ -376,9 +376,9 @@ TEE_Result stm32_bsec_shadow_read_otp(uint32_t *val, uint32_t otp)
 	if (otp > bsec_dev.max_id)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	/* for non-secure OTP: only return the shadow value */
-	if (!(bsec_dev.shadow->status[otp] & PTA_BSEC_STATUS_SECURE)) {
-		*val = bsec_dev.shadow->value[otp];
+	/* for non-secure OTP: only return the mirror value */
+	if (!(bsec_dev.mirror->status[otp] & PTA_BSEC_STATUS_SECURE)) {
+		*val = bsec_dev.mirror->value[otp];
 
 		return TEE_SUCCESS;
 	}
@@ -430,15 +430,15 @@ TEE_Result stm32_bsec_write_otp(uint32_t val, uint32_t otp)
 		io_write32(bsec_base() + BSEC_FVR(otp), val);
 
 		bsec_unlock(exceptions);
-	} else if ((bsec_dev.shadow->status[otp] & PTA_BSEC_STATUS_SECURE)) {
+	} else if ((bsec_dev.mirror->status[otp] & PTA_BSEC_STATUS_SECURE)) {
 		/* update of secure and not-shadowed OTP is not allowed */
 		return TEE_ERROR_ACCESS_DENIED;
 	}
 
-	/* Update the shadow memory when it is allowed */
-	if (!(bsec_dev.shadow->status[otp] & PTA_BSEC_STATUS_SECURE)) {
-		bsec_dev.shadow->value[otp] = val;
-		bsec_dev.shadow->status[otp] &= ~PTA_BSEC_LOCK_ERROR;
+	/* Update the mirror memory when it is allowed */
+	if (!(bsec_dev.mirror->status[otp] & PTA_BSEC_STATUS_SECURE)) {
+		bsec_dev.mirror->value[otp] = val;
+		bsec_dev.mirror->status[otp] &= ~PTA_BSEC_LOCK_ERROR;
 	}
 
 	return TEE_SUCCESS;
@@ -484,8 +484,8 @@ TEE_Result stm32_bsec_program_otp(uint32_t val, uint32_t otp)
 
 	exceptions = bsec_lock();
 
-	bsec_dev.shadow->value[otp] = 0x0U;
-	bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_ERROR;
+	bsec_dev.mirror->value[otp] = 0x0U;
+	bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_ERROR;
 
 	for (i = 0U; i < MAX_NB_TRIES; i++) {
 		uint32_t status = 0U;
@@ -530,10 +530,10 @@ TEE_Result stm32_bsec_program_otp(uint32_t val, uint32_t otp)
 			EMSG("BSEC shadow %"PRIu32" invalid: %08x, write= %08x",
 			     otp, fvr, val);
 
-		/* update the shadow memory if allowed */
-		if (!(bsec_dev.shadow->status[otp] & PTA_BSEC_STATUS_SECURE)) {
-			bsec_dev.shadow->value[otp] = fvr;
-			bsec_dev.shadow->status[otp] &= ~PTA_BSEC_LOCK_ERROR;
+		/* update the mirror memory if allowed */
+		if (!(bsec_dev.mirror->status[otp] & PTA_BSEC_STATUS_SECURE)) {
+			bsec_dev.mirror->value[otp] = fvr;
+			bsec_dev.mirror->status[otp] &= ~PTA_BSEC_LOCK_ERROR;
 		}
 	}
 
@@ -652,7 +652,7 @@ TEE_Result stm32_bsec_set_sr_lock(uint32_t otp)
 		 * even if the lock is activated.
 		 */
 		io_write32(bsec_base() + BSEC_SRLOCK(bank), bank_value);
-		bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_SHADOW_R;
+		bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_SHADOW_R;
 		result = TEE_SUCCESS;
 	}
 
@@ -668,7 +668,7 @@ TEE_Result stm32_bsec_read_sr_lock(uint32_t otp, bool *value)
 	if (otp > bsec_dev.max_id)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	*value = bsec_dev.shadow->status[otp] & PTA_BSEC_LOCK_SHADOW_R;
+	*value = bsec_dev.mirror->status[otp] & PTA_BSEC_LOCK_SHADOW_R;
 
 	return TEE_SUCCESS;
 }
@@ -710,7 +710,7 @@ TEE_Result stm32_bsec_set_sw_lock(uint32_t otp)
 		 * even if the lock is activated.
 		 */
 		io_write32(bsec_base() + BSEC_SWLOCK(bank), bank_value);
-		bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_SHADOW_W;
+		bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_SHADOW_W;
 		result = TEE_SUCCESS;
 	}
 
@@ -726,7 +726,7 @@ TEE_Result stm32_bsec_read_sw_lock(uint32_t otp, bool *value)
 	if (otp > bsec_dev.max_id)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	*value = bsec_dev.shadow->status[otp] & PTA_BSEC_LOCK_SHADOW_W;
+	*value = bsec_dev.mirror->status[otp] & PTA_BSEC_LOCK_SHADOW_W;
 
 	/*
 	 * OEM keys are accessible only in ROM code
@@ -770,7 +770,7 @@ TEE_Result stm32_bsec_set_sp_lock(uint32_t otp)
 		 * even if the lock is activated.
 		 */
 		io_write32(bsec_base() + BSEC_SPLOCK(bank), bank_value);
-		bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_SHADOW_P;
+		bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_SHADOW_P;
 		result = TEE_SUCCESS;
 	}
 
@@ -786,7 +786,7 @@ TEE_Result stm32_bsec_read_sp_lock(uint32_t otp, bool *value)
 	if (otp > bsec_dev.max_id)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	*value = bsec_dev.shadow->status[otp] & PTA_BSEC_LOCK_SHADOW_P;
+	*value = bsec_dev.mirror->status[otp] & PTA_BSEC_LOCK_SHADOW_P;
 
 	return TEE_SUCCESS;
 }
@@ -828,7 +828,7 @@ TEE_Result stm32_bsec_permanent_lock_otp(uint32_t otp)
 	}
 
 	if (!result)
-		bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_PERM;
+		bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_PERM;
 
 	bsec_unlock(exceptions);
 
@@ -842,17 +842,17 @@ TEE_Result stm32_bsec_read_permanent_lock(uint32_t otp, bool *value)
 	if (otp > bsec_dev.max_id)
 		return TEE_ERROR_BAD_PARAMETERS;
 
-	*value = bsec_dev.shadow->status[otp] & PTA_BSEC_LOCK_PERM;
+	*value = bsec_dev.mirror->status[otp] & PTA_BSEC_LOCK_PERM;
 
 	return TEE_SUCCESS;
 }
 
 bool stm32_bsec_can_access_otp(uint32_t otp)
 {
-	if (bsec_dev.shadow->magic != BSEC_MAGIC)
+	if (bsec_dev.mirror->magic != BSEC_MAGIC)
 		return false;
 
-	if (bsec_dev.shadow->state == BSEC_STATE_INVALID)
+	if (bsec_dev.mirror->state == BSEC_STATE_INVALID)
 		return false;
 
 	if (otp < bsec_dev.upper_base)
@@ -862,7 +862,7 @@ bool stm32_bsec_can_access_otp(uint32_t otp)
 		return false;
 
 	/* manage HIDEUP */
-	if ((bsec_dev.shadow->status[otp] & HIDEUP_ERROR) == HIDEUP_ERROR)
+	if ((bsec_dev.mirror->status[otp] & HIDEUP_ERROR) == HIDEUP_ERROR)
 		return false;
 
 	if (otp > bsec_dev.max_id)
@@ -873,10 +873,10 @@ bool stm32_bsec_can_access_otp(uint32_t otp)
 
 bool stm32_bsec_nsec_can_access_otp(uint32_t otp)
 {
-	if (bsec_dev.shadow->magic != BSEC_MAGIC)
+	if (bsec_dev.mirror->magic != BSEC_MAGIC)
 		return false;
 
-	if (bsec_dev.shadow->state == BSEC_STATE_INVALID)
+	if (bsec_dev.mirror->state == BSEC_STATE_INVALID)
 		return false;
 
 	/* lower OTP are accessible by non secure */
@@ -890,13 +890,13 @@ bool stm32_bsec_nsec_can_access_otp(uint32_t otp)
 		return false;
 
 	/* manage HIDEUP */
-	if ((bsec_dev.shadow->status[otp] & HIDEUP_ERROR) == HIDEUP_ERROR)
+	if ((bsec_dev.mirror->status[otp] & HIDEUP_ERROR) == HIDEUP_ERROR)
 		return false;
 
 	/* upper OTP with provisioning tag and not locked are accessible */
-	if ((bsec_dev.shadow->status[otp] & PTA_BSEC_STATUS_PROVISIONING) ==
+	if ((bsec_dev.mirror->status[otp] & PTA_BSEC_STATUS_PROVISIONING) ==
 	    PTA_BSEC_STATUS_PROVISIONING &&
-	    (bsec_dev.shadow->status[otp] & PTA_BSEC_LOCK_PERM) !=
+	    (bsec_dev.mirror->status[otp] & PTA_BSEC_LOCK_PERM) !=
 	    PTA_BSEC_LOCK_PERM)
 		return true;
 
@@ -907,8 +907,8 @@ TEE_Result stm32_bsec_get_state(enum stm32_bsec_sec_state *state)
 {
 	assert(state);
 
-	if (bsec_dev.shadow->magic == BSEC_MAGIC)
-		*state = bsec_dev.shadow->state;
+	if (bsec_dev.mirror->magic == BSEC_MAGIC)
+		*state = bsec_dev.mirror->state;
 	else
 		*state = BSEC_STATE_INVALID;
 
@@ -933,7 +933,7 @@ static uint32_t init_state(uint32_t status)
 			EMSG("BSEC invalid nvstates %#x\n", nvstates);
 		} else {
 			state = BSEC_STATE_SEC_OPEN;
-			if (bsec_dev.shadow->value[OTP_SECURE_BOOT] &
+			if (bsec_dev.mirror->value[OTP_SECURE_BOOT] &
 			    OTP_CLOSED_SECURE)
 				state = BSEC_STATE_SEC_CLOSED;
 		}
@@ -942,7 +942,7 @@ static uint32_t init_state(uint32_t status)
 	return state;
 }
 
-static void stm32_bsec_shadow_load(uint32_t status)
+static void stm32_bsec_mirror_load(uint32_t status)
 {
 	unsigned int otp = 0U, bank = 0U;
 	uint32_t exceptions = 0U;
@@ -952,9 +952,9 @@ static void stm32_bsec_shadow_load(uint32_t status)
 	uint32_t mask = 0U;
 	unsigned int max_id = bsec_dev.max_id;
 
-	memset(bsec_dev.shadow, 0, sizeof(*bsec_dev.shadow));
-	bsec_dev.shadow->magic = BSEC_MAGIC;
-	bsec_dev.shadow->state = BSEC_STATE_INVALID;
+	memset(bsec_dev.mirror, 0, sizeof(*bsec_dev.mirror));
+	bsec_dev.mirror->magic = BSEC_MAGIC;
+	bsec_dev.mirror->state = BSEC_STATE_INVALID;
 
 	exceptions = bsec_lock();
 
@@ -962,8 +962,8 @@ static void stm32_bsec_shadow_load(uint32_t status)
 	if (status & BSEC_OTPSR_HIDEUP) {
 		for (otp = bsec_dev.upper_base;
 		     otp <= bsec_dev.max_id ; otp++) {
-			bsec_dev.shadow->status[otp] |= HIDEUP_ERROR;
-			bsec_dev.shadow->value[otp] = 0x0U;
+			bsec_dev.mirror->status[otp] |= HIDEUP_ERROR;
+			bsec_dev.mirror->value[otp] = 0x0U;
 		}
 		max_id = bsec_dev.upper_base - 1;
 	}
@@ -979,13 +979,13 @@ static void stm32_bsec_shadow_load(uint32_t status)
 		mask = otp_bit(otp);
 
 		if (srlock[bank] & mask)
-			bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_SHADOW_R;
+			bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_SHADOW_R;
 		if (swlock[bank] & mask)
-			bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_SHADOW_W;
+			bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_SHADOW_W;
 		if (splock[bank] & mask)
-			bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_SHADOW_P;
+			bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_SHADOW_P;
 
-		if (bsec_dev.shadow->status[otp] & PTA_BSEC_STATUS_SECURE)
+		if (bsec_dev.mirror->status[otp] & PTA_BSEC_STATUS_SECURE)
 			continue;
 
 		/*
@@ -993,30 +993,30 @@ static void stm32_bsec_shadow_load(uint32_t status)
 		 * They are stored in last OTPs
 		 */
 		if (otp >= OEM_KEY_FIRST_OTP) {
-			bsec_dev.shadow->status[otp] |= PTA_BSEC_LOCK_SHADOW_R;
+			bsec_dev.mirror->status[otp] |= PTA_BSEC_LOCK_SHADOW_R;
 			continue;
 		}
 
 		/* reload shadow to read Permanent Programing Lock Flag */
 		shadow_otp(otp);
 
-		bsec_dev.shadow->value[otp] = io_read32(bsec_base() +
+		bsec_dev.mirror->value[otp] = io_read32(bsec_base() +
 					      BSEC_FVR(otp));
 	}
 
 	bsec_unlock(exceptions);
 }
 
-static void stm32_bsec_shadow_init(void)
+static void stm32_bsec_mirror_init(void)
 {
 	uint32_t status = bsec_get_otp_status();
 
 	/* update bsec miror */
-	stm32_bsec_shadow_load(status);
+	stm32_bsec_mirror_load(status);
 
 	/* always update status */
-	bsec_dev.shadow->state = init_state(status);
-	if (bsec_dev.shadow->state == BSEC_STATE_INVALID)
+	bsec_dev.mirror->state = init_state(status);
+	if (bsec_dev.mirror->state == BSEC_STATE_INVALID)
 		panic("BSEC invalid state");
 }
 
@@ -1159,14 +1159,15 @@ static void initialize_nvmem_layout_from_dt(void *fdt, int bsec_node)
 		if (fdt_getprop(fdt, node, "st,secure-otp", NULL)) {
 			for (otp = cell->otp_id;
 			     otp < cell->otp_id + otp_nb; otp++)
-				bsec_dev.shadow->status[otp] |= PTA_BSEC_STATUS_SECURE;
+				bsec_dev.mirror->status[otp] |=
+					PTA_BSEC_STATUS_SECURE;
 		}
 		/* check if provisioning is allowed */
 		else if (fdt_getprop(fdt, node,
 				     "st,non-secure-otp-provisioning", NULL)) {
 			for (otp = cell->otp_id;
 			     otp < cell->otp_id + otp_nb; otp++)
-				bsec_dev.shadow->status[otp] |=
+				bsec_dev.mirror->status[otp] |=
 					PTA_BSEC_STATUS_PROVISIONING;
 		}
 	}
@@ -1203,8 +1204,8 @@ stm32_bsec_pm(enum pm_op op, unsigned int pm_hint,
 		/* treat BSEC error */
 		check_reset_error();
 
-		/* re initialize the shadow */
-		stm32_bsec_shadow_init();
+		/* re initialize the mirror */
+		stm32_bsec_mirror_init();
 	}
 
 	return TEE_SUCCESS;
@@ -1220,15 +1221,15 @@ static TEE_Result initialize_bsec(void)
 	plat_bsec_get_static_cfg(&cfg);
 
 	if (IS_ENABLED(CFG_STM32_CM33TDCID))
-		va = core_mmu_get_va(cfg.shadow, MEM_AREA_RAM_NSEC, SIZE_4K);
+		va = core_mmu_get_va(cfg.mirror, MEM_AREA_RAM_NSEC, SIZE_4K);
 	else
-		va = core_mmu_get_va(cfg.shadow, MEM_AREA_RAM_SEC, SIZE_4K);
+		va = core_mmu_get_va(cfg.mirror, MEM_AREA_RAM_SEC, SIZE_4K);
 
 	if (!va)
 		return TEE_ERROR_GENERIC;
 
 	bsec_dev.base.pa = cfg.base;
-	bsec_dev.shadow = (struct bsec_shadow *)va;
+	bsec_dev.mirror = (struct bsec_mirror *)va;
 	bsec_dev.upper_base = cfg.upper_start;
 	bsec_dev.max_id = cfg.max_id;
 
@@ -1240,7 +1241,7 @@ static TEE_Result initialize_bsec(void)
 	initialize_bsec_from_dt(fdt, node);
 
 	if (IS_ENABLED(CFG_STM32_CM33TDCID)) {
-		if (bsec_dev.shadow->magic != BSEC_MAGIC)
+		if (bsec_dev.mirror->magic != BSEC_MAGIC)
 			panic("BSEC mirror is not initialized\n");
 	} else {
 		if ((IS_ENABLED(CFG_STM32MP21) &&
@@ -1252,8 +1253,8 @@ static TEE_Result initialize_bsec(void)
 
 		check_reset_error();
 
-		/* initialize the shadow */
-		stm32_bsec_shadow_init();
+		/* initialize the mirror */
+		stm32_bsec_mirror_init();
 
 		register_pm_core_service_cb(stm32_bsec_pm, NULL, "stm32-bsec");
 	}
diff --git a/core/include/drivers/stm32_bsec.h b/core/include/drivers/stm32_bsec.h
index 7c5505ee2..9feb60028 100644
--- a/core/include/drivers/stm32_bsec.h
+++ b/core/include/drivers/stm32_bsec.h
@@ -35,13 +35,13 @@ enum stm32_bsec_sec_state {
  * Structure and API function for BSEC driver to get some platform data.
  *
  * @base: BSEC interface registers physical base address
- * @shadow: BSEC shadow base address
+ * @mirror: BSEC mirror base address
  * @upper_start: Base ID for the BSEC upper words in the platform
  * @max_id: Max value for BSEC word ID for the platform
  */
 struct stm32_bsec_static_cfg {
 	paddr_t base;
-	paddr_t shadow;
+	paddr_t mirror;
 	unsigned int upper_start;
 	unsigned int max_id;
 };
-- 
2.47.2

